### PR TITLE
PB-32034 [Signer UI] - Allow customizing NSC logo via .Net SDK

### PR DESCRIPTION
**Pull request by sohanal1**
Allow customizing NSC logo via .Net SDK

### CODE REVIEW

File: sdk/SDK.Examples/SDK.Examples.csproj [hunk 1]
```diff
@@ -222,6 +223,9 @@
     <None Include="src\Text_Tag_And_Acrobat_Form_Fields.pdf">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
+    <None Include="signers.properties">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
```
Comment (SeungChoi) [inline sdk/SDK.Examples/SDK.Examples.csproj]: Why did you add this?

File: sdk/SDK.Examples/src/SigningLogoExample.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
...ykrH4LZR7YCbSXvwS6gY/AfRGbx4gvBOPBKpAD/gqeAsz1hUJKhkq/aAdmmjz9IYJKX3BdOFbzRytkBj/0G1AMskH4qzeE04GKkvJv4LdAiVG7IZHGaS87CHaBb4GOpk97iF4PTg9SyZ8JJoJnwWmw2LQlrt/v1x6m7dIDfxxkga+AXp4PR/w+Q1x76DEwDSg5s4H+c4KHlJQ8cNSs2C0k/2dAZSwwSVlut0X/MFAZAS4H2oP0DFuXFvFLwXGwzixLXHvLZ0BFxxY9iKnA2uBMXEnRPN0pZQlpJUF74T+pknAj0KHA7tMg/SZQsZNSRdqxVkVa2/uhbsXS7drx1lrD1zt2VkOJvwvsYu5v74NSnWkZ3Uw4A0cElnxJbCS6LtxhzoD9pL9OWsvdCW4Hz4FJ6PS71Dpvq6J3wfU10Zq/pPdg8z4pvcQzCfVXak+htxxqHHkG/QPALofI221XEN8Evquk6G03g4LXYLSDuJvcEVFuIVQSVU7VBNZfk1glrmdEO8nKdUYqSV4O/maq8a17d/9XQVt099Mu8DiYBVT0Vu0meSh1LDN9ajpa9PEiestZQbQR6XJwkQ5kMyOZOjjpATkE3RMotNsuo+H7HJneCSVPRb/52s4Df6Pub4P/rjGt/Yv9CVJ6IN1qteHelGxL6+Wil4n2LPWnZ11PTAaIlrtQhAft6AzPNJXPByq7gQ46/cCtYC5Q0QEyvHeN0BxTxtorRK9lVB4FbcE2oONVeHclof4a4+dAB9rwFivC4UAHt9fAAKB9XccdHUxVvq4+CHVMUdFeFO6phDpI6+BdBPpG7J4nngtuMNtG/CagA6rKwxE7HWgdl2JEPwV9zWeUiKgDNdTBMVpWo+hncxKXlIiDKymndxWVHKB3DpVlIMVWn94lCjUDUYJUSsA0m42SMgfsAGqbCXTA1oMdbbNT4l8GOihvBnpiNP5LcBT4IsUapSMHMoBwHOgPikEm0O9BWwMXFetPTx4EH6E/SBgW9KlEHgV86abmo1XoLiL9j+BGoK8KfyFvBaFDsOuEQm16Ab1stmIX/vUr8XBPIVgI5oGRQPfe6c9iVmFnDsQka4T6hhEbAXQcycRmH7r7ie8kfpj4IOJ82dhYS2gJej1u5aDlC43VM/4vTdXSxAafpmpVM/tpJcXlBPw/No3coziOudIAAAAASUVORK5CYII=\"}]";
```
Comment (SeungChoi) [inline sdk/SDK.Examples/src/SigningLogoExample.cs]: This is completely different from Java SDK example. You need to synchronize two.

File: sdk/Silanis.ESL.SDK/src/Models/SigningLogo.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+        [JsonProperty ("language")]
+        public String Language {
+            get; set;
+        }
+
+        [JsonProperty ("image")]
+        public String Image {
+            get; set;
+        }
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/SigningLogo.cs]: Please put "{" on the next line so for the C # formatting.

File: sdk/Silanis.ESL.SDK/src/Services/SigningLogoService.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            try {
+                restClient.Post (path, "[]");
+            } catch (EslServerException e) {
+                throw new EslServerException ("Could not delete the signing logos from account." + " Exception: " + e.Message, e.ServerError, e);
+            } catch (Exception e) {
+                throw new EslException ("Could not delete the signing logos from account." + " Exception: " + e.Message, e);
+            }
+        }
+
+    }
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Services/SigningLogoService.cs]: Please put "{" on the next line so for the C # formatting.

File: sdk/Silanis.ESL.SDK/src/SigningLogo.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        }
+
+        public string Language {
+            get; private set;
+        }
+
+        public string Image {
+            get; private set;
+        }
+    }
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/SigningLogo.cs]: Please put "{" on the next line so for the C # formatting.

File: sdk/Silanis.ESL.SDK/src/Services/SigningLogoService.cs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            string path = template.UrlFor (UrlTemplate.ACCOUNT_SIGNING_LOGO_PATH)
+                          .Build ();
+
+            try {
+                restClient.Post (path, "[]");
+            } catch (EslServerException e) {
+                throw new EslServerException ("Could not delete the signing logos from account." + " Exception: " + e.Message, e.ServerError, e);
+            } catch (Exception e) {
+                throw new EslException ("Could not delete the signing logos from account." + " Exception: " + e.Message, e);
+            }
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Services/SigningLogoService.cs]: The `create`, `delete`, and `update` methods are actually the same. Please unify it into one method.

File: sdk/Silanis.ESL.SDK/src/Services/SigningLogoService.cs (hunk 4 above)
Comment (sohanal1) [reply sdk/Silanis.ESL.SDK/src/Services/SigningLogoService.cs]: done

File: sdk/Silanis.ESL.SDK/src/SigningLogo.cs (hunk 5 above)
Comment (sohanal1) [reply sdk/Silanis.ESL.SDK/src/SigningLogo.cs]: done

File: sdk/Silanis.ESL.SDK/src/Internal/Conversion/SigningLogoConverter.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        public SigningLogoConverter (Silanis.ESL.SDK.SigningLogo sdkSigningLogo)
+        {
+            this.sdkSigningLogo = sdkSigningLogo;
+        }
+
+        public Silanis.ESL.API.SigningLogo ToAPISigningLogo ()
+        {
+            if (sdkSigningLogo == null) {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Internal/Conversion/SigningLogoConverter.cs]: To match the C # code style, start { with the following line.

File: sdk/Silanis.ESL.SDK/src/Internal/Conversion/SigningLogoConverter.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+            Silanis.ESL.API.SigningLogo result = new Silanis.ESL.API.SigningLogo ();
+            result.Language = sdkSigningLogo.Language;
+            result.Image = sdkSigningLogo.Image;
+
+            return result;
+        }
+
+        public Silanis.ESL.SDK.SigningLogo ToSDKSigningLogo ()
+        {
+            if (apiSigningLogo == null) {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Internal/Conversion/SigningLogoConverter.cs]: To match the C # code style, start { with the following line.

File: sdk/Silanis.ESL.SDK/src/Services/SigningStyleService.cs [hunk 9]
```diff
@@ -71,11 +84,53 @@ public void DeleteSigningThemes ()
...
             }
         }
+
+        public void SaveSigningLogos (List<SigningLogo> signingLogos)
+        {
+            string path = template.UrlFor (UrlTemplate.ACCOUNT_SIGNING_LOGO_PATH)
+                          .Build ();
+            string payload = JsonConvert.SerializeObject (signingLogos, new JsonSerializerSettings { ContractResolver = new CamelCasePropertyNamesContractResolver (), Formatting = Formatting.Indented });
+
+            try {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Services/SigningStyleService.cs]: To match the C # code style, start { with the following line.

File: sdk/Silanis.ESL.SDK/src/Services/SigningStyleService.cs [hunk 10]
```diff
@@ -71,11 +84,53 @@ public void DeleteSigningThemes ()
...
+                throw new EslException ("Could not save the signing logos for account." + " Exception: " + e.Message, e);
+            }
+        }
+
+        public List<SigningLogo> GetSigningLogos ()
+        {
+            string path = template.UrlFor (UrlTemplate.ACCOUNT_SIGNING_LOGO_PATH)
+                          .Build ();
+
+            try {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Services/SigningStyleService.cs]: To match the C # code style, start { with the following line.

File: sdk/Silanis.ESL.SDK/src/Internal/Conversion/SigningLogoConverter.cs (hunk 7 above)
Comment (sohanal1) [reply sdk/Silanis.ESL.SDK/src/Internal/Conversion/SigningLogoConverter.cs]: done

File: sdk/Silanis.ESL.SDK/src/Internal/Conversion/SigningLogoConverter.cs (hunk 8 above)
Comment (sohanal1) [reply sdk/Silanis.ESL.SDK/src/Internal/Conversion/SigningLogoConverter.cs]: done

File: sdk/Silanis.ESL.SDK/src/Services/SigningStyleService.cs (hunk 9 above)
Comment (sohanal1) [reply sdk/Silanis.ESL.SDK/src/Services/SigningStyleService.cs]: done
